controllers/topology/internal/scope/upgradetracker.go

```diff
@@ -0,0 +1,45 @@
+/*
```

Comment:
I am inclined to leave it in a separate file. It would be nice to have the separation and would be useful as we would be implementing some upgrade strategies and we can use this file to have the upgrade code in one place.

Reply:
Fine for me